plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/InjectedProjection.java

```diff
@@ -43,11 +46,16 @@ public List<String> getProjectedValues(Optional<Domain> partitionValueFilter)
         Domain domain = partitionValueFilter
                 .orElseThrow(() -> new InvalidProjectionException(columnName, "Injected projection requires single predicate for it's column in where clause"));
         Type type = domain.getType();
-        if (!domain.isNullableSingleValue() || !canConvertSqlTypeToStringForParts(type, true)) {
+        if (!(type instanceof VarcharType || type instanceof CharType)) {
```

Comment:
Limiting to `Varchar` and `CharType` would be a step backwards for previous usages that were relying on the semantics of the previous check that did `!canConvertSqlTypeToStringForParts(type, true)`. The Athena docs are indeed out of date on this point- we also constrain the types allowed by checking `canConvertSqlTypeToStringForParts(type, true)`

Reply:
Thank you for confirming and for the helpful point!